### DISCUSSION TITLE
How to serve nextjs app using nginx in production using docker

### DISCUSSION THREAD

**Discussion by qburst-sachin** [vercel/next.js#22018] [Help]
Currently am working on app in which nextjs is used for frontend app. 
As intial prototype is ready I intend to make a production release. 

So as part of it while dockerizing the nextjs app. I used multi staged build.

First I ran npm run build and generated the static content. 

Then I built nginx layer over it and copied the contents from /out folder to nginx root folder from previous build phase.

And I was able to serve the site currently it ran as expected.

So I want to ask is this the right approach if I have to serve the content using nginx or is there any other approach.

**Comment by mkrtchian** [comment]
Your option is ok if your frontend app is fully static, as you did `next export` and are serving files generated in `out/` folder.

This means that you will not be able to execute server-side code with Next.js, like [Server Side Rendering](https://nextjs.org/docs/basic-features/pages#server-side-rendering), [Incremental Static Regeneration](https://nextjs.org/docs/basic-features/data-fetching#incremental-static-regeneration), and some functionalities will not be available like [Image optimization](https://github.com/vercel/next.js/discussions/19065).

To have all the functionalities available, you could start the Next.js server with `next start` in your docker container, and redirect traffic to its port (by default 3000) with nginx.
